planner/core/stats.go

```diff
@@ -202,7 +203,17 @@ func (ts *TableScan) DeriveStats(childStats []*property.StatsInfo) (_ *property.
 	}
 	ts.Source.deriveStatsByFilter(ts.AccessConds)
 	sc := ts.SCtx().GetSessionVars().StmtCtx
-	ts.Ranges, err = ranger.BuildTableRange(ts.AccessConds, sc, ts.Handle.RetType)
+	if ts.Handle != nil {
+		ts.Ranges, err = ranger.BuildTableRange(ts.AccessConds, sc, ts.Handle.RetType)
+	} else {
```

Comment:
When would `ts.Handle` be nil? According to `Convert2Gathers -> buildTableGather -> ds.getHandleCol`, it can never be nil when coming to here?

Reply:
`ds.getHanldeCol` may return nil.
```go
func (ds *DataSource) getHandleCol() *expression.Column {
	if ds.handleCol != nil {
		return ds.handleCol
	}

	if !ds.tableInfo.PKIsHandle {
		ds.handleCol = ds.newExtraHandleSchemaCol()
		return ds.handleCol
	}

	for i, col := range ds.Columns {
		if mysql.HasPriKeyFlag(col.Flag) {
			ds.handleCol = ds.schema.Columns[i]
			break
		}
	}

	return ds.handleCol
}
```
for sql `select b from t`, with primary key `a`.
Because I introduce `PruneColumn` before, so `ds.handleCol` is nil and `ds.tableInfo.PKIsHandle` is true, and `ds.Columns` doesn't contain `a`, so that ds.handleCol is nil finally.

Reply:
OK, please add this comment here for explanation.